src/deluge/util/functions.cpp

```diff
@@ -933,43 +933,163 @@ FilterType stringToFilterType(char const* string) {
+	switch (oldMode) {
+	case OldArpMode::DOWN:
+		return ArpNoteMode::DOWN;
+	case OldArpMode::RANDOM:
+		return ArpNoteMode::RANDOM;
+	default:
+		return ArpNoteMode::UP;
+	}
+}
+
+ArpOctaveMode oldModeToArpOctaveMode(OldArpMode oldMode) {
+	switch (oldMode) {
+	case OldArpMode::DOWN:
+		return ArpOctaveMode::DOWN;
+	case OldArpMode::BOTH:
+		return ArpOctaveMode::ALTERNATE;
+	case OldArpMode::RANDOM:
+		return ArpOctaveMode::RANDOM;
+	default:
+		return ArpOctaveMode::UP;
+	}
+}
+
+char const* oldArpModeToString(OldArpMode mode) {
 	switch (mode) {
-	case ArpMode::UP:
+	case OldArpMode::UP:
 		return "up";
 
-	case ArpMode::DOWN:
+	case OldArpMode::DOWN:
 		return "down";
 
-	case ArpMode::BOTH:
+	case OldArpMode::BOTH:
 		return "both";
 
-	case ArpMode::RANDOM:
+	case OldArpMode::RANDOM:
 		return "random";
 
 	default:
 		return "off";
 	}
 }
 
-ArpMode stringToArpMode(char const* string) {
+OldArpMode stringToOldArpMode(char const* string) {
 	if (!strcmp(string, "up")) {
-		return ArpMode::UP;
+		return OldArpMode::UP;
 	}
 	else if (!strcmp(string, "down")) {
-		return ArpMode::DOWN;
+		return OldArpMode::DOWN;
 	}
 	else if (!strcmp(string, "both")) {
-		return ArpMode::BOTH;
+		return OldArpMode::BOTH;
 	}
 	else if (!strcmp(string, "random")) {
-		return ArpMode::RANDOM;
+		return OldArpMode::RANDOM;
+	}
+	else {
+		return OldArpMode::OFF;
+	}
+}
+
+char const* arpModeToString(ArpMode mode) {
+	switch (mode) {
+	case ArpMode::ARP:
+		return "arp";
+
+	default:
+		return "off";
+	}
+}
+
+ArpMode stringToArpMode(char const* string) {
+	if (!strcmp(string, "arp")) {
+		return ArpMode::ARP;
 	}
 	else {
 		return ArpMode::OFF;
 	}
 }
 
+char const* arpNoteModeToString(ArpNoteMode mode) {
+	switch (mode) {
+	case ArpNoteMode::DOWN:
+		return "down";
+
+	case ArpNoteMode::UP_DOWN:
+		return "upDown";
+
+	case ArpNoteMode::AS_PLAYED:
+		return "asPlayed";
+
+	case ArpNoteMode::RANDOM:
+		return "random";
+
+	default:
+		return "up";
+	}
+}
+
+ArpNoteMode stringToArpNoteMode(char const* string) {
+	if (!strcmp(string, "down")) {
+		return ArpNoteMode::DOWN;
+	}
+	else if (!strcmp(string, "upDown")) {
+		return ArpNoteMode::UP_DOWN;
+	}
+	else if (!strcmp(string, "asPlayed")) {
+		return ArpNoteMode::AS_PLAYED;
+	}
+	else if (!strcmp(string, "random")) {
+		return ArpNoteMode::RANDOM;
+	}
+	else {
+		return ArpNoteMode::UP;
+	}
+}
+
+char const* arpOctaveModeToString(ArpOctaveMode mode) {
+	switch (mode) {
+	case ArpOctaveMode::DOWN:
+		return "down";
+
+	case ArpOctaveMode::RANDOM:
+		return "random";
+
+	default:
+		return "up";
+	}
+}
+
+ArpOctaveMode stringToArpOctaveMode(char const* string) {
+	if (!strcmp(string, "down")) {
+		return ArpOctaveMode::DOWN;
+	}
+	else if (!strcmp(string, "upDown")) {
+		return ArpOctaveMode::RANDOM;
+	}
+	else if (!strcmp(string, "alt")) {
+		return ArpOctaveMode::ALTERNATE;
+	}
+	else if (!strcmp(string, "random")) {
+		return ArpOctaveMode::RANDOM;
+	}
+	else {
+		return ArpOctaveMode::UP;
+	}
+}
+
```

Comment:
These can/should be replaced with `EnumStringMap` I think, should provide some code size savings (see `src/deluge/model/mod_controllable/filters/filter_config.cpp`  for an example of how to use these). They can live in `arpeggiator.cpp` as static functions too, so `functions.cpp` doesn't expand forever.

Reply:
I specify different names for OLED and 7SEG so probably not a good fit this EnumStringMap

Reply:
I'm working on converting params to use the enumStringMap as a bigger refactor, don't worry about it for now